src/ralph/networks/filters.py

```diff
@@ -77,3 +80,45 @@ def queryset(self, request, queryset):
         elif self.value().lower() == 'public':
             queryset = queryset.exclude(PRIVATE_NETWORK_FILTER)
         return queryset
+
+
+class ContainsIPAddressFilter(TextListFilter):
+
+    title = _('Contains IP address')
+    parameter_name = 'contains_ip'
+
+    def __init__(self, field, request, params, model, model_admin, field_path):
+        super(ContainsIPAddressFilter, self).__init__(
+            field, request, params, model, model_admin, field_path
+        )
+        self.title = _('Contains IP address')
+
+    def queryset(self, request, queryset):
+        filter_str = self.value()
+        if not filter_str:
+            return queryset
+
+        # Replace all possible separators with spaces
+        filter_str = filter_str.translate(
+            str.maketrans(
+                SEARCH_OR_SEPARATORS,
+                ' ' * len(SEARCH_OR_SEPARATORS)
+            )
+        )
+
+        filter_query = Q()
+
+        for str_addr in filter_str.split(' '):
+            try:
+                address = int(ipaddress.ip_address(str_addr))
+            except ValueError:
+                return queryset
```

Comment:
you could add warning message in such case. See [here](https://github.com/allegro/ralph/blob/ng/src/ralph/admin/filters.py#L351) for example.